Area51/Sources/ViewModels/FeedViewModel.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import ListingService
+
+class FeedViewModel: BaseViewModel {
+
+    var subredditName: String?
+    var updated: (() -> Void)?
+    private let subreddit: Subreddit
+    private var listings = [Listing]()
+
+    var listingsCount: Int {
+        return listings.count
+    }
+
+    private var dataSource: ListingsDataSource! {
+        didSet {
+            self.dataSource.updated = { [weak self] in
+                if let listings = self?.dataSource.listings {
+                    self?.listings = listings
+                    self?.removeHtmlFromListings()
+                }
+
+                self?.updated?()
+            }
+        }
+    }
+
+    override func didBecomeActive() {
+        dataSource = ListingsDataSource(subreddit: subreddit)
+    }
+
```

Comment:
I feel using this `didBecomeActive` approach is a good way to 'tell' the model when to start an operation. Be it, fetching data or computing some other operation. The means in the `ViewController` all we need to do is set the corresponding `ViewModel`'s `isActive` to true at a point where we want the model to be active and thus, start fetching data.  

Setting this value to true again, without changing it to false first, will not call 'didBecomeActive()'  again. We might want to set it to false when navigating away from a screen and need the screen to refresh automatically when returning to it; so in this case, we set `viewModel.isActive = false` in `viewWillDisappear` or something similar and then `viewModel.isActive = true` in `viewDidAppear`

Reply:
Why does a viewModel need to have state for being active or not active?  In this case, you could move `dataSource = ListingsDataSource(subreddit: subreddit)` inside of `init(with subreddit: Subreddit)`. What is gained by setting active to true or false in viewWillAppear / viewWillDisappear?

Reply:
Having the data source set in the init would mean the network call is made right away. It might be too early for that. Generally I think having it set to active after all views have been configured correctly and ready to receive data is a better approach. 

As I mentioned in the previous comment; this gives us the freedom to decide when do start the request for data from the server.

Reply:
I see. I think we should simplify the architecture unless we actually need the additional complexity. In this case, the ViewController is calling bind when it is ready to receive the callback (viewDidLoad) so there is no harm in loading the data on initialization. I would actually prefer to load the data sooner than later since it will increase how fast the user is able to view the data.

Removing isActive would also remove the need for a base class. In general we should keep state to a minimum unless needed.

Reply:
Fair enough. I’ll make the changes and update it within the next hour when I get back to my PC.

Reply:
dope thanks!

Reply:
Updated as per our discussion. A note - I had to use defer in the initialiser, otherwise didSet will not be called on the datasource as it is too early in the initialisation process. 

I think this way works just as well as my original implementation and for more complex setups - we can relook at it.